Build issues on NixOS

After probably https://github.com/tree-sitter/tree-sitter/commit/f9e8802234cbd692b1979cff3bea90f5103d4187 I run into

```
warning: In file included from /nix/store/3w4yrspgcx1z9wi18rv0hbap4jjlhx4c-glibc-2.37-8-dev/include/bits/libc-header-start.h:33,
warning:                  from /nix/store/3w4yrspgcx1z9wi18rv0hbap4jjlhx4c-glibc-2.37-8-dev/include/stdlib.h:26,
warning:                  from include/tree_sitter/api.h:10,
warning:                  from src/./alloc.h:4,
warning:                  from src/./alloc.c:1,
warning:                  from src/lib.c:8:
warning: /nix/store/3w4yrspgcx1z9wi18rv0hbap4jjlhx4c-glibc-2.37-8-dev/include/features.h:413:4: error: #warning _FORTIFY_SOURCE requires compiling with optimization (-O) [-Werror=cpp]
warning:   413 | #  warning _FORTIFY_SOURCE requires compiling with optimization (-O)
warning:       |    ^~~~~~~
warning: cc1: all warnings being treated as errors
```

This is because NixOS enables it by default (`-O2 -D_FORTIFY_SOURCE=2`) and the build seems to override the level to 0: 

```
running: "gcc" "-O0" "-ffunction-sections" "-fdata-sections" "-fPIC" "-m64" "-I" "src" "-I" "include" "-Wall" "-Wextra" "-std=c99" "-fvisibility=hidden" "-Wshadow" "-Werror" "-o" "src/helix/target/re
lease/build/tree-sitter-0070f96918a4c153/out/src/lib.o" "-c" "src/lib.c"
```

This was hidden away before by `Werror` being disabled.

\cc @amaanq

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -177,6 +177,7 @@ jobs:
         TARGET: ${{ matrix.target }}
         CC: ${{ matrix.cc }}
         AR: ${{ matrix.ar }}
+        IS_WINDOWS: ${{ contains(matrix.os, 'windows') }}
       run: |
         PATH="$PWD/.github/scripts:$PATH"
         echo "$PWD/.github/scripts" >> $GITHUB_PATH
@@ -191,6 +192,8 @@ jobs:
         [ -n "$CC" ] && echo "CC=$CC" >> $GITHUB_ENV
         [ -n "$AR" ] && echo "AR=$AR" >> $GITHUB_ENV
 
+        [ "$IS_WINDOWS" = "false" ] && echo "CFLAGS=-Werror" >> $GITHUB_ENV
+
         if [ "$USE_CROSS" == "true" ]; then
           echo "BUILD_CMD=cross" >> $GITHUB_ENV
           runner=$(BUILD_CMD=cross cross.sh bash -c "env | sed -nr '/^CARGO_TARGET_.*_RUNNER=/s///p'")
@@ -199,7 +202,7 @@ jobs:
 
     - name: Build C library
       if: ${{ !contains(matrix.os, 'windows') }} # Requires an additional adapted Makefile for `cl.exe` compiler
-      run: make.sh CFLAGS="-Werror" -j
+      run: make.sh -j
 
     - name: Build wasm library
       if: ${{ !matrix.cli-only && !matrix.use-cross }} # No sense to build on the same Github runner hosts many times

--- a/Makefile
+++ b/Makefile
@@ -18,7 +18,7 @@ endif
 OBJ := $(SRC:.c=.o)
 
 # define default flags, and override to append mandatory flags
-override CFLAGS := -O3 -std=gnu99 -fPIC -fvisibility=hidden -Wall -Wextra -Werror -Wshadow $(CFLAGS)
+override CFLAGS := -O3 -std=gnu99 -fPIC -fvisibility=hidden -Wall -Wextra -Wshadow $(CFLAGS)
 override CFLAGS += -Ilib/src -Ilib/include
 
 # ABI versioning

--- a/lib/binding_rust/build.rs
+++ b/lib/binding_rust/build.rs
@@ -28,7 +28,6 @@ fn main() {
         .flag_if_supported("-std=c99")
         .flag_if_supported("-fvisibility=hidden")
         .flag_if_supported("-Wshadow")
-        .flag_if_supported("-Werror")
         .include(src_path)
         .include("include")
         .file(src_path.join("lib.c"))

